File: src/pages/introduction-to-graphs/adjacency_list.py
class AdjacencyList:
    # An Adjacency List for storing directed, unweighted graphs
    n = 0
    lst = None

    def __init__(self, numberNodes):
        # Creates a new instance for n nodes, indexed 0
        self.lst = [[] for i in range(numberNodes)]
        self.n = numberNodes

    def getEdge(self, u, v):
        # Gets the edge between nodes u and v, if possible
        assert(u>=0 and u<self.n and v>=0 and v<self.n)
        return v in self.lst[u]

    def addEdge(self, u, v):
        # Adds an edge between nodes u and v, if possible
        assert(u>=0 and u<self.n and v>=0 and v<self.n)
        self.lst[u].append(v)

    def removeEdge(self, u, v):
        # Removes the edge between nodes u and v, if possible
        assert(u>=0 and u<self.n and v>=0 and v<self.n)
        self.lst[u].remove(v)

    def printGraph(self):
        # Prints graph, if possible
        for i in range(self.n):
            assert(self.lst[i] != None)
            print(i,  ":", self.lst[i])

File: src/pages/articles/introduction-to-graphs/AdjacencyList.java
class AdjacencyList {
    // An Adjacency List for storing directed, unweighted graphs
    int n;
    ArrayList<Integer>[] lst;

    public AdjacencyList(int numberNodes){
        // Creates a new instance for n nodes, indexed 0
        this.lst = new ArrayList[numberNodes];
        for(int i = 0; i<numberNodes; ++i){
            this.lst[i] = new ArrayList<Integer>();
        }
        this.n = numberNodes;
    }

    public boolean getEdge(int u, int v){
        // Gets the edge between nodes u and v, if possible
        assert(u>=0 && u<this.n && v>=0 && v<this.n);
        return this.lst[u].contains(v);
    }

    public void addEdge(int u, int v){
        // Adds an edge between nodes u and v, if possible
        assert(u>=0 && u<this.n && v>=0 && v<this.n);
        this.lst[u].add(v);
    }

    public void removeEdge(int u, int v){
        // Removes the edge between nodes u and v, if possible
        assert(u>=0 && u<this.n && v>=0 && v<this.n);
        this.lst[u].remove(Integer.valueOf(v));
    }
    
    public void printGraph(){
        // Prints graph, if possible
        for(int i = 0; i<this.n; ++i){
            assert(this.lst[i] != null);
            System.out.println(Integer.toString(i) + ": " + this.lst[i].toString());
        }
    }
}    